File: contest/LeetCode_Weekly_Contest_206/RabbitLee/Special_Positions_in_a_Binary_Matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        rows, cols = len(mat), len(mat[0])
        row_list = [sum(row) for row in mat]
        col_list = []
        for j in range(cols):
            col_list.append(sum([row[j] for row in mat]))
        
        res = 0
        for i in range(rows):
            for j in range(cols):
                if mat[i][j] == 1:
                    if row_list[i] == 1 and col_list[j] == 1:
                        res += 1
        return res

File: contest/LeetCode_Weekly_Contest_206/Special_Positions_in_a_Binary_Matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        int n = mat.size();
        int m = mat.front().size();
        
        int ret = 0;
        for (int i = 0; i < n; ++i) 
            for (int j = 0; j < m; ++j) if (mat[i][j]) {
                bool flag = true;
                for (int c = 0; c < m; ++c) if (c != j && mat[i][c]) {
                    flag = false;
                    break;
                }
                if (!flag) continue;
                for (int r = 0; r < n; ++r) if (r != i && mat[r][j]) {
                    flag = false;
                    break;
                }
                if (flag) ret++;
            }
        return ret;
    }
};
